backend/src/test/java/com/google/sps/EmailNotifierTest.java

```diff
@@ -0,0 +1,178 @@
+import org.mockito.Mock;
+
+public class EmailNotifierTest {
+  private EmailNotifier emailNotifier;
+  @Mock private Gmail gmail;
+  @Mock private Gmail.Users users;
+  @Mock private Gmail.Users.Messages messages;
+  @Mock private Gmail.Users.Messages.Send send;
+  private Message testMessage;
+
+  @Before
+  public void setUp() throws MessagingException, IOException {
+    gmail = mock(Gmail.class);
+    // Mock to return list of users
+    users = mock(Gmail.Users.class);
+    // Mock to return list of messages.
+    messages = mock(Gmail.Users.Messages.class);
+    send = mock(Gmail.Users.Messages.Send.class);
+    MimeMessage mimemessage = createTestMessage();
+    testMessage = createMessageFromMime(mimemessage);
+    // Test Instance of email notifier class.
+    emailNotifier = new EmailNotifier("John", "jdoe@gmail.com ", gmail);
+    // Emulating real behavior due to when method, will return list of users.
+    when(gmail.users()).thenReturn(users);
+    // Emulating real behaviour due to when method, will return list of messages
+    when(users.messages()).thenReturn(messages);
+    //    Emulating sending messages, when sent will return send mock.
+    when(messages.send(anyString(), any(Message.class))).thenReturn(send);
+    //    When send mock is executed will return test message.
+    when(send.execute()).thenReturn(testMessage);
+  }
+
+  public MimeMessage createTestMessage() throws MessagingException {
+    Properties properties = System.getProperties();
+    Session session = Session.getDefaultInstance(properties);
+    MimeMessage message = new MimeMessage(session);
+    Address[] addresses =
+        new Address[] {
+          new InternetAddress("tcwang@google.com"),
+          new InternetAddress("grantjustice@google.com"),
+          new InternetAddress("kevinhowald@google.com")
+        };
+    message.setFrom(new InternetAddress("Adlib-Step@gmail.com"));
+    message.setSubject("Test Subject");
+    message.setText("Test Text");
+    message.setRecipients(javax.mail.Message.RecipientType.TO, addresses);
+    return message;
+  }
+
+  public Message createMessageFromMime(MimeMessage mimeMessage)
+      throws IOException, MessagingException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    mimeMessage.writeTo(buffer);
+    byte[] bytes = buffer.toByteArray();
+    String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+    Message message = new Message();
+    message.setRaw(encodedEmail);
+    return message;
+  }
+
+  public MimeMessage convertToMimeMessage(Message msg) throws MessagingException {
+    Properties properties = System.getProperties();
+    Session session = Session.getDefaultInstance(properties);
+    byte[] bytes = msg.decodeRaw();
+    InputStream targetStream = new ByteArrayInputStream(bytes);
+    return new MimeMessage(session, targetStream);
+  }
+
+  @Test
+  public void constructorShouldSetNameAndEmail() {
+    assertEquals("John", emailNotifier.recipientName);
```

Comment:
I think I commented earlier in the PR but I can't find it so apologies if I missed a response. Please convert all assertions to [Truth](www.truth.dev)

Reply:
Ack